sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/WebSocketsProxyConnectionHandler.java

```diff
@@ -26,57 +24,78 @@
-    private final String amqpHostname;
-    private final String remoteHost;
+    private final InetSocketAddress connectionHostname;
     private final ProxyOptions proxyOptions;
+    private final String fullyQualifiedNamespace;
+    private final String amqpBrokerHostname;
 
     /**
      * Creates a handler that handles proton-j's connection through a proxy using web sockets.
+     * The hostname of the proxy is exposed in {@link #getHostname()}.
      *
      * @param connectionId Identifier for this connection.
-     * @param amqpHostname Hostname of the AMQP message broker. The hostname of the proxy is exposed in {@link
-     *     #getHostname()}.
      * @param proxyOptions The options to use for proxy.
-     * @param product The name of the product this connection handler is created for.
+     * @param productName The name of the product this connection handler is created for.
      * @param clientVersion The version of the client library creating the connection handler.
-     * @param clientOptions provided by the user.
+     * @param connectionOptions Options used when creating the connection.
      *
      * @throws NullPointerException if {@code amqpHostname} or {@code proxyConfiguration} is null.
+     * @throws IllegalStateException if a proxy address is unavailable for the given {@code proxyOptions}.
      */
-    public WebSocketsProxyConnectionHandler(String connectionId, String amqpHostname, ProxyOptions proxyOptions,
-        String product, String clientVersion, SslDomain.VerifyMode verifyMode, ClientOptions clientOptions) {
-        super(connectionId, amqpHostname, product, clientVersion, verifyMode, clientOptions);
+    public WebSocketsProxyConnectionHandler(String connectionId, String productName, String clientVersion,
+        ConnectionOptions connectionOptions, ProxyOptions proxyOptions) {
+        super(connectionId, productName, clientVersion, connectionOptions);
 
-        this.amqpHostname = Objects.requireNonNull(amqpHostname, "'amqpHostname' cannot be null.");
         this.proxyOptions = Objects.requireNonNull(proxyOptions, "'proxyConfiguration' cannot be null.");
-        this.remoteHost = amqpHostname + ":" + HTTPS_PORT;
+        this.fullyQualifiedNamespace = connectionOptions.getFullyQualifiedNamespace();
+        this.amqpBrokerHostname = connectionOptions.getFullyQualifiedNamespace() + ":" + connectionOptions.getPort();
+
+        if (proxyOptions.isProxyAddressConfigured()) {
+            this.connectionHostname = (InetSocketAddress) proxyOptions.getProxyAddress().address();
+        } else {
+            final URI serviceUri = createURI(connectionOptions.getHostname(), connectionOptions.getPort());
+            final ProxySelector proxySelector = ProxySelector.getDefault();
+            if (proxySelector == null) {
+                throw logger.logExceptionAsError(new IllegalStateException("ProxySelector should not be null."));
```

Comment:
Should this be NPE instead?

Reply:
I don't believe so because this is not a parameter given by users.. I think it would imply that something in the system went wrong.